frontend/src/App.tsx

```diff
@@ -555,9 +555,23 @@ export class App extends PureComponent<Props, State> {
         )
       }
 
+      // NOTE: This code (and the !themeInput case below) needs to be fixed to
+      // handle the case where the user has their theme preference set to
+      // Auto/Dark/Light and we don't want to ignore that.
       if (themeInput.setAsDefault) {
         this.props.theme.setTheme(customTheme)
       }
+    } else if (!themeInput) {
```

Comment:
I would add an and condition here for if the current theme is a custom theme. We only need to run this then I believe? Then I believe it'll solve the handling user preference business?

Reply:
Oh yeah, will do. It'll fix it for the theme being removed case, but I don't think there's an easy solution for the add case